apiserver/facades/client/charms/client.go

```diff
@@ -178,11 +177,7 @@ func (a *API) getDownloadInfo(ctx context.Context, arg params.CharmURLAndOrigin)
 }
 
 func (a *API) getDefaultArch() (string, error) {
-	cons, err := a.backendState.ModelConstraints()
```

Comment:
We need to replace this with a call to the domain service.

Reply:
This logic will be on the application epic to decide how this is going to re-implemented in their relevant domains. @CodingCookieRookie  can you please add a TODO here for this work that needs to be done. JUJU-7622 has also been created to track this.